geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -0,0 +1,286 @@
+    }
+    sizeInBytes = InternalDataSerializer.readPrimitiveInt(in);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof RedisSortedSet)) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+    RedisSortedSet redisSortedSet = (RedisSortedSet) o;
+    if (members.size() != redisSortedSet.members.size()) {
+      return false;
+    }
+
+    for (Map.Entry<byte[], byte[]> entry : members.entrySet()) {
+      if (!Arrays.equals(redisSortedSet.members.get(entry.getKey()), (entry.getValue()))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public int getDSFID() {
+    return REDIS_SORTED_SET_ID;
+  }
+
+  protected synchronized AddOrChange memberAdd(byte[] memberToAdd, byte[] scoreToAdd) {
+    boolean added = (members.put(memberToAdd, scoreToAdd) == null);
+    if (added) {
+      sizeInBytes += calculateSizeOfNewFieldValuePair(memberToAdd, scoreToAdd);
+    }
+    return added ? AddOrChange.ADDED : AddOrChange.NOOP;
+  }
+
+  private synchronized void membersAddAll(AddsDeltaInfo addsDeltaInfo) {
+    Iterator<byte[]> iterator = addsDeltaInfo.getAdds().iterator();
+    while (iterator.hasNext()) {
+      byte[] member = iterator.next();
+      sizeInBytes += PER_PAIR_OVERHEAD + member.length;
+      byte[] score = iterator.next();
+      sizeInBytes += PER_PAIR_OVERHEAD + score.length;
+      members.put(member, score);
+    }
+  }
+
+
+  private synchronized void membersRemoveAll(RemsDeltaInfo remsDeltaInfo) {
+    for (byte[] member : remsDeltaInfo.getRemoves()) {
+      sizeInBytes -= (PER_PAIR_OVERHEAD + member.length);
+      members.remove(member);
+    }
+  }
+
+  /**
+   * @param region the region this instance is stored in
+   * @param key the name of the set to add to
+   * @param membersToAdd members to add to this set; NOTE this list may by modified by this call
+   * @return the number of members actually added
+   */
+  long zadd(Region<RedisKey, RedisData> region, RedisKey key, List<byte[]> membersToAdd,
+      SortedSetOptions options) {
+    int membersAdded = 0;
+    if (options == null) {
+      options = new SortedSetOptions(SortedSetOptions.Exists.NONE, SortedSetOptions.Update.NONE);
+    }
+    AddsDeltaInfo deltaInfo = null;
+    Iterator<byte[]> iterator = membersToAdd.iterator();
+    while (iterator.hasNext()) {
+      boolean delta = true;
+      byte[] score = iterator.next();
+      byte[] member = iterator.next();
+
+      if (options.isNX()) {
+        if (members.get(member) != null) {
+          continue;
+        }
+      }
+      if (options.isXX()) {
+        if (members.get(member) == null) {
+          continue;
+        }
+      }
+      switch (memberAdd(member, score)) {
+        case ADDED:
+          membersAdded++;
+          makeAddsDeltaInfo(deltaInfo, member, score);
+          break;
+        case CHANGED:
+          // TODO: track changed
```

Comment:
What does this comment mean? Is there an existing ticket that relates to it?

Reply:
Just the Tracker story "Implement ZADD option CH". May have gotten ahead of ourselves when we put that in...